### ISSUE TITLE
Unbundle LuaJIT

### ISSUE THREAD

**Issue by tim77** [howl-editor/howl#503]
Please make it possible to build Howl with LuaJIT with system version, not bundled version.

https://docs.fedoraproject.org/en-US/packaging-guidelines/#bundling

**Comment by refi64** [comment]
Hmm Fedora doesn't seem to build LuaJIT with Lua 5.2 support, which Howl uses (not sure how extensively). Maybe that would fall under:

> All packages whose upstreams have no mechanism to build against system libraries MAY opt to carry bundled libraries

**Comment by tim77** [comment]
Nice @refi64 Maybe possible in this case to build with bundled then. Review request is there btw https://bugzilla.redhat.com/show_bug.cgi?id=1701204Thank you.

**Comment by peterljung** [comment]
I am wondering if that would be an option for OpenBSD as well.

E.g. OpenBSD 6.5 has packages for:
* lua-lpeg-1.0.1.tgz flavours: lua52 and lua53
* luajit-2.0.5p1.tgz

Howl doesn't seem compile with default clang compiler currently on amd64.
I have to force it to use gcc. It seems to be related to libunwind but I don't understand exactly why changing compiler makes it work.

Following works.
```
gmake CC=gcc
gmake install
```
But I would like to have a package at some point ...

**Comment by nilnor** [comment]
Howl do use the 5.2 compatibility (I don't have a ready list of all usage however). As for OpenBSD, the 2.0 line of LuaJIT is pretty ancient by now - it's an unfortunate fact that 2.1 is stuck in beta, I believe that a lot of people actually use 2.1 for the production needs (e.g. OpenResty). Both things together makes the unbundling situation kind of a hard solve unfortunately.

**Comment by ignatenkobrain** [comment]
Maintainer of Fedora's luajit here.

> Hmm Fedora doesn't seem to build LuaJIT with Lua 5.2 support, which Howl uses (not sure how extensively). Maybe that would fall under:

What do you mean? Is there some flag to enable it? We ship 2.1.0-beta3 with huge patchsets on top of that to make s390x and ppc64 stuff work.

**Comment by refi64** [comment]
@ignatenkobrain Indeed, Howl passes [-DLUAJIT_ENABLE_LUA52_COMPAT](https://luajit.org/extensions.html#lua52), which enables several changes that would be breaking (e.g. the write return type, removal of gfind). As far as I can tell, the LuaJIT spec file in Fedora does [not pass that](https://src.fedoraproject.org/rpms/luajit/blob/master/f/luajit.spec#_130).

Some of those patches look pretty interesting, kinda sad that LuaJIT is in a really awkward state right now wrt maintenance and build-time configuration.

**Comment by ignatenkobrain** [comment]
Fair enough. https://src.fedoraproject.org/rpms/luajit/pull-request/2

**Comment by refi64** [comment]
Nice, if the breaking changes don't have any negative impact then that's pretty much perfect afaik.

**Comment by daurnimator** [comment]
Could Howl use https://github.com/lunarmodules/lua-compat-5.3 to work with any lua/luajit?

**Comment by nilnor** [comment]
> Could Howl use https://github.com/lunarmodules/lua-compat-5.3 to work with any lua/luajit?

No, it relies heavily on LuaJIT's FFI.

**Comment by nilnor** [comment]
Sorry for the very long silence here.

Howl development has been quiet on `master` for years, while work continued on the `next` branch — that branch has now been merged. Master has changed substantially since the 0.6 release: a full port to GTK 4, and a rewrite of the command line and interaction APIs, among a lot else.

We're closing out the old backlog, and feature requests are being closed for good rather than parked. Howl is developed at a pace that doesn't support an open wishlist, and leaving these sitting here indefinitely implies a roadmap that doesn't exist. That's not a comment on the merit of the idea — thank you for taking the time to write it up.

Please don't reopen this one. Bug reports against current master are still welcome.
